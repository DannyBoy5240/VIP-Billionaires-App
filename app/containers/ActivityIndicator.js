import React from 'react'
import { ActivityIndicator, StyleSheet } from 'react-native'
import { themes } from '../constants/colors'

const styles = StyleSheet.create({
  indicator: {
    padding: 16,
    flex: 1,
  },
  absolute: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: 100000,
  },
})

const RCActivityIndicator = ({ theme, absolute, ...props }) => (
  <ActivityIndicator
    style={[styles.indicator, absolute && styles.absolute]}
    color={themes[theme].activeTintColor}
    {...props}
  />
)
export default RCActivityIndicator
