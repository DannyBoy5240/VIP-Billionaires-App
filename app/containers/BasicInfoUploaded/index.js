import React from 'react'
import { View, Text, Image } from 'react-native'

import styles from './style'
import { VectorIcon } from '../VectorIcon'
import images from '../../assets/images'
import { themes } from '../../constants/colors'
import { useTheme } from '../../theme'

const BasicInfoUploaded = ({ name, gender, dob, phone, location, showCloseIcon }) => {

  const { theme } = useTheme()

  return (
    <View style={styles.container}>
      <View>
        <Text style={[styles.name, { color: themes[theme].titleColor }]}>{name}</Text>
        <Text style={[styles.genderAndDob, { color: themes[theme].titleColor }]}>
          {gender} | {dob}
        </Text>
        <View style={styles.iconAndText}>
          <VectorIcon
            type="MaterialIcons"
            name="perm-phone-msg"
            color="#F5BF4D"
            size={18}
          />
          <Text style={[styles.phone, { color: themes[theme].titleColor }]}>{phone}</Text>
        </View>
        <View style={styles.iconAndText}>
          <Image
            source={images.location_home}
            style={styles.locationHomeImage}
          />
          <Text style={[styles.location, { color: themes[theme].titleColor }]}>{location}</Text>
        </View>
      </View>
      {showCloseIcon && (
        <VectorIcon
          type="Ionicons"
          name="close-outline"
          size={20}
          color="#858585"
          style={styles.closeIcon}
        />
      )}
    </View>
  );
}

export default BasicInfoUploaded