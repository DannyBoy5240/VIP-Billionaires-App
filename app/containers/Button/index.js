import React from 'react'
import { StyleSheet, View, Text, Image, ActivityIndicator, TouchableOpacity } from 'react-native'
import { RectButton } from 'react-native-gesture-handler'
import { COLOR_BLACK, COLOR_BTN_BACKGROUND, COLOR_BTN_BORDER, COLOR_WHITE, themes } from '../../constants/colors'
import images from '../../assets/images'
import { VectorIcon } from '../VectorIcon'

/* eslint-disable react-native/no-unused-styles */
const styles = StyleSheet.create({
  container: {
    borderRadius: 13,
    backgroundColor: COLOR_WHITE,
  },
  oauthButtonContainer: {
    margin: 8,
  },
  oauthBtn: {
    height: 40,
    resizeMode: 'contain',
  },
  buttonContainer: {
    borderRadius: 7,
    height: 45,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: 16,
    borderWidth: 0.25,
  },
  icon: {
    position: 'absolute',
    left: 7.5,
    top: 7.5,
    width: 35,
    height: 35,
    resizeMode: 'contain',
  },
  text: {
    fontSize: 16,
    textAlign: 'center',
    fontFamily: 'Raleway',
    lineHeight: 18,
  },

  button_size_Z: {
    width: 335,
  },
  button_size_Y: {
    width: 201,
  },
  button_size_X: {
    width: 125,
  },
  button_size_W: {
    width: '100%',
  },
  button_size_V: {
    width: 125,
  },
  button_size_U: {
    width: 140,
  },
  button_size_T: {
    width: 313,
  },
  button_size_S: {
    width: 345,
  },
})

const Button = props => {
  const {
    title,
    text,
    icon,
    type,
    size,
    onPress,
    iconCenter,
    hidden,
    disabled,
    width,
    height,
    backgroundColor,
    loading,
    style,
    textColor: color,
    fontWeight,
    theme,
    iconColor,
    ...otherProps
  } = props

  if (hidden) {
    return null
  }

  if (type === 'oauth') {
    return (
      <RectButton onPress={onPress} style={styles.oauthButtonContainer}>
        <Image source={images[title]} style={styles.oauthBtn} />
      </RectButton>
    )
  }

  if (type === 'secondary') {
    return (
      <TouchableOpacity
        onPress={onPress}
        disabled={(disabled || loading)}
        style={[
          styles.container,
          size ? styles[`button_size_${size}`] : {},
          style,
        ]}
        {...otherProps}>
        <View
          style={[
            styles.buttonContainer,
            size ? styles[`button_size_${size}`] : {},
            width ? { width } : {},
            height ? { height } : {},
            {
              opacity: disabled ? 0.5 : 1,
              borderWidth: 0.5,
              backgroundColor: COLOR_WHITE,
              borderColor: COLOR_BTN_BORDER,
            },
          ]}>
          {icon ? (
            <VectorIcon
              type="MaterialCommunityIcons" name={icon}
              size={24} style={{ marginRight: 8, ...iconColor }} />
          ) : null}
          {loading ? (
            <ActivityIndicator color={COLOR_BLACK} />
          ) : (
            <>
              <Text style={[styles.text, { color: disabled ? themes[theme].textColor : COLOR_BLACK }]}>
                {text || title}
              </Text>
            </>
          )}
        </View>
      </TouchableOpacity>
    )
  }

  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={(disabled || loading)}
      style={[
        styles.container,
        size ? styles[`button_size_${size}`] : {},
        style,
      ]}
      {...otherProps}>
      <View
        style={[
          styles.buttonContainer,
          size ? styles[`button_size_${size}`] : {},
          width ? { width } : {},
          height ? { height } : {},
          {
            backgroundColor: COLOR_BTN_BACKGROUND,
            borderColor: COLOR_BTN_BORDER,
          },
        ]}>
        {icon ? (
          <VectorIcon
            type="MaterialCommunityIcons" name={icon}
            size={24} style={{ marginRight: 8, ...iconColor }} />
        ) : null}
        {loading ? (
          <ActivityIndicator color={COLOR_WHITE} />
        ) : (
          <>
            <Text style={[styles.text, { color: disabled ? themes[theme].textColor : COLOR_WHITE }]}>
              {text || title}
            </Text>
          </>
        )}
      </View>
    </TouchableOpacity>
  )
}

Button.defaultProps = {
  title: 'Press me!',
  type: 'primary',
  size: 'Z',
  onPress: () => alert('It works!'),
  icon: null,
  hidden: false,
  disabled: false,
  loading: false,
}

export default Button
