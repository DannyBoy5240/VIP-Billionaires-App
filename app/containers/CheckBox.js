import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import Ionicons from 'react-native-vector-icons/Ionicons'
import { COLOR_BLACK_LIGHT, COLOR_BLUE } from '../constants/colors'

const CheckBox = React.memo(
  ({
     title,
     checked,
     checkedColor,
     unCheckedColor,
     checkedIcon,
     uncheckedIcon,
     textStyle,
     containerStyle,
     onPress,
   }) => (
    <TouchableOpacity
      style={{
        ...containerStyle,
        flexDirection: 'row',
        alignItems: 'center',
        height: 49,
        paddingHorizontal: 12,
        borderWidth: 1,
        borderRadius: 11,
        borderColor: checked ? COLOR_BLUE : COLOR_BLACK_LIGHT,
        marginBottom: 5,
      }}
      onPress={onPress}>
      <Ionicons
        name={checked ? checkedIcon : uncheckedIcon}
        size={12}
        color={checked ? checkedColor : unCheckedColor}
      />
      {title ? (
        <Text style={{ marginLeft: 8, ...textStyle }}>{title}</Text>
      ) : null}
    </TouchableOpacity>
  ),
)
export default CheckBox
