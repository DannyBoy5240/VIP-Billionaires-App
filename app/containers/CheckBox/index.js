import React from 'react'
import { Text, TouchableOpacity } from 'react-native'
import { COLOR_YELLOW } from '../../constants/colors'
import { VectorIcon } from '../VectorIcon'

const Index = React.memo(
  ({ title, checked, textStyle, containerStyle, onPress }) => (
    <TouchableOpacity
      style={{
        ...containerStyle,
        flexDirection: 'row',
        alignItems: 'center',
        paddingVertical: 8,
      }}
      onPress={onPress}>
      <VectorIcon
        name={checked ? 'check-square-o' : 'square-o'}
        type={'FontAwesome'}
        size={24}
        color={checked ? COLOR_YELLOW : 'black'}
      />
      {title ? (
        <Text style={{ marginLeft: 8, ...textStyle }}>{title}</Text>
      ) : null}
    </TouchableOpacity>
  ),
)

export default Index
