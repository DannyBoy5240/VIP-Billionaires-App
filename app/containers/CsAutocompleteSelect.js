import React, {useState} from 'react';
import {Dimensions, StyleSheet, View, Text} from 'react-native';
import {AutocompleteDropdown} from 'react-native-autocomplete-dropdown';
import {themes} from '../constants/colors';
import {VectorIcon} from './VectorIcon';

const {width, height} = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    width: width * 0.92,
    borderWidth: 1,
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'center',
    paddingLeft: 10,
    borderRadius: 8,
    paddingHorizontal: 10,
    justifyContent: 'space-between',
    height: 56,
    borderColor: '#C4C4C4',
    marginBottom: 15,
  },
  label: {
    fontFamily: 'Raleway',
    fontWeight: '400',
    color: '#C4C4C4',
    marginBottom: 5,
    fontSize: 14,
  },
});

const CsAutocompleteSelect = ({
  leftIcon,
  data,
  onSelectItem,
  placeholder,
  theme,
  label,
}) => {
  const [show, setShow] = useState(false);

  const RightIcon = () => {
    return (
      <VectorIcon
        type={'Entypo'}
        name={show ? 'chevron-thin-up' : 'chevron-thin-right'}
        color={themes[theme].activeTintColor}
        size={18}
      />
    );
  };

  return (
    <>
      {label && (
        <Text style={[styles.label, themes[theme].normalTextColor]}>
          {label}
        </Text>
      )}
      <View style={styles.container}>
        <AutocompleteDropdown
          clearOnFocus={false}
          closeOnBlur={true}
          closeOnSubmit={false}
          dataSet={data}
          onSelectItem={onSelectItem}
          textInputProps={{
            placeholder: placeholder,
            autoCorrect: false,
            autoCapitalize: 'none',
            style: {
              fontFamily: 'Raleway',
              color: themes[theme].activeTintColor,
              fontSize: 14,
            },
          }}
          inputContainerStyle={{
            backgroundColor: 'transparent',
            alignSelf: 'flex-end',
          }}
          suggestionsListContainerStyle={{
            width: width * 0.76,
            marginLeft: -40,
            backgroundColor: themes[theme].backgroundColor,
          }}
          containerStyle={{width: '95%'}}
          showChevron={false}
        />
        <RightIcon />
      </View>
    </>
  );
};

export default CsAutocompleteSelect;
