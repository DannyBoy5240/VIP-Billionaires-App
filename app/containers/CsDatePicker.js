import React, { useState } from 'react';
import { StyleSheet, Text, TouchableOpacity, Platform, View } from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import moment from 'moment';
import { DATE_STRING_FORMAT, TIME_STRING_FORMAT } from '../utils/datetime';
import { themes } from '../constants/colors';
import sharedStyles from '../views/Styles';
import { isIOS } from '../utils/deviceInfo';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: 'grey',
    marginBottom: 10,
  },
  label: {
    fontSize: 14,
    width: 80,
    paddingLeft: 8,
    paddingVertical: 4,
    ...sharedStyles.textSemibold,
  },
  input: {
    ...sharedStyles.textRegular,
    fontSize: 16,
    paddingHorizontal: isIOS ? 8 : 14,
    paddingVertical: 4,
    flexGrow: 1,
  },
});

export default function CsDatePicker({
  label,
  containerStyle,
  placeholder,
  style,
  type,
  value,
  action,
  minimumDate,
  theme,
}) {
  const [show, onShow] = useState(false);
  const [currentDate, onChangeDate] = useState(value);

  const onChange = ({ nativeEvent: { timestamp } }, date) => {
    if (Platform.OS === 'android') {
      onShow(false);
    }
    if (type === 'date') {
      console.log('date', date);
      if (!date) {
        return;
      }
      const newDate = date;
      onChangeDate(newDate);
      action({ value: moment(newDate).format(DATE_STRING_FORMAT) });
    } else {
      console.log('time', timestamp);
      if (!timestamp) {
        return;
      }
      const newDate = new Date(timestamp);
      onChangeDate(newDate);
      action({ value: moment(newDate).format(TIME_STRING_FORMAT) });
    }
  };

  // const getTimeString = date => {
  //   let hours = date.getHours();
  //   let minutes = date.getMinutes();
  //   let ampm = hours >= 12 ? 'PM' : 'AM';
  //   hours = hours % 12;
  //   hours = hours ? hours : 12; // the hour '0' should be '12'
  //   minutes = minutes < 10 ? '0' + minutes : minutes;
  //   let strTime = hours + ':' + minutes + ' ' + ampm;
  //   return strTime;
  // };

  if (Platform.OS === 'ios') {
    return (
      <View style={[styles.container, containerStyle]}>
        {label ? (
          <Text
            contentDescription={null}
            accessibilityLabel={null}
            style={[styles.label, { color: themes[theme].infoText }]}>
            {label}
          </Text>
        ) : null}
        <DateTimePicker
          testID='date-time-picker'
          style={{ width: 116 }}
          mode={type}
          is24Hour={true}
          display="default"
          minimumDate={minimumDate}
          value={currentDate ?? new Date()}
          onChange={onChange}
          textColor={'black'}
          themeVariant={'light'}
        />
      </View>
    );
  }

  let button = (
    <TouchableOpacity style={styles.input} onPress={() => onShow(!show)}>
      <Text>
        {currentDate
          ? moment(currentDate).format(
            type === 'date' ? DATE_STRING_FORMAT : TIME_STRING_FORMAT,
          )
          : placeholder}
      </Text>
    </TouchableOpacity>
  );

  const content = show ? (
    <DateTimePicker
      style={style}
      mode={type}
      display="default"
      minimumDate={minimumDate}
      value={currentDate ?? new Date()}
      onChange={onChange}
      textColor={'black'}
    />
  ) : null;

  return (
    <View style={[styles.container, containerStyle]}>
      {label ? (
        <Text
          contentDescription={null}
          accessibilityLabel={null}
          style={[styles.label, { color: themes[theme].infoText }]}>
          {label}
        </Text>
      ) : null}
      {button}
      {content}
    </View>
  );
}
