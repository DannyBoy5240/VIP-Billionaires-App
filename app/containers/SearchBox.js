import React, { useState } from 'react'
import { View, StyleSheet, TextInput, Text, TouchableOpacity } from 'react-native'
import sharedStyles from '../views/Styles'
import { withTheme } from '../theme'
import { themes, COLOR_TRANSPARENT, COLOR_GRAY_DARK } from '../constants/colors'
import { VectorIcon } from './VectorIcon'

const styles = StyleSheet.create({
  container: {
    borderStyle: 'solid',
    borderColor: COLOR_GRAY_DARK,
    borderWidth: 0.5,
    borderRadius: 14,
  },
  searchBox: {
    alignItems: 'center',
    borderRadius: 24,
    flexDirection: 'row',
    fontSize: 17,
    paddingHorizontal: 10,
    marginVertical: 4,
    marginHorizontal: 8,
  },
  input: {
    fontFamily: 'Hind Vadodara',
    height: 40,
    flex: 1,
    fontSize: 16,
    marginLeft: 8,
    paddingTop: 0,
    paddingBottom: 0,
    ...sharedStyles.textRegular,
  },
  cancel: {
    padding: 4,
  },
  cancelText: {
    ...sharedStyles.textRegular,
    fontSize: 17,
    backgroundColor: 'red',
  },
})

export const SearchBox = ({
  onChangeText,
  onSubmitEditing,
  testID,
  hasCancel,
  onCancelPress,
  inputRef,
  placeholder,
  theme,
  clearTextType,
  ...props
}) => {
  const [isShowClear, setShowClear] = useState(false)
  const [text, setText] = useState('')
  const onChangeTextInput = (_text) => {
    setText(_text)
    setShowClear(_text && _text.length > 0)
    onChangeText(_text)
  }
  const clear = () => {
    setText('')
    setShowClear(false)
    onChangeText('')
  }

  return (
    <View style={styles.container}>
      <View style={[styles.searchBox, { backgroundColor: COLOR_TRANSPARENT }]}>
        <VectorIcon
          name="search"
          type="Ionicons"
          size={18}
          color={themes[theme].subTextColor}
        />
        <TextInput
          ref={inputRef}
          value={text}
          autoCapitalize="none"
          autoCorrect={false}
          blurOnSubmit
          placeholder={placeholder ?? 'Search'}
          returnKeyType="search"
          style={[styles.input, { color: themes[theme].titleColor }]}
          testID={testID}
          underlineColorAndroid="transparent"
          onChangeText={onChangeTextInput}
          onSubmitEditing={onSubmitEditing}
          placeholderTextColor={themes[theme].textColor}
          theme={theme}
          {...props}
        />
        {
          isShowClear
            ? (<TouchableOpacity onPress={clear}>
                {
                  clearTextType
                    ? (<Text style={{ color: themes[theme].activeTintColor }}>Clear</Text>)
                    : (<VectorIcon name="close" type="Ionicons" size={18} color={themes[theme].activeTintColor} />)
                }
              </TouchableOpacity>
            ) : (<></>)
        }
      </View>
    </View>
  )
}

export default withTheme(SearchBox)
