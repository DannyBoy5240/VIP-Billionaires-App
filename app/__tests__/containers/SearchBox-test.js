import React from 'react'
import renderer from 'react-test-renderer'
import { SearchBox } from '../../containers/SearchBox'

test('renders correctly', () => {
  const tree = renderer.create(<SearchBox onChangeText={() => {}} theme="dark" />).toJSON()
  expect(tree).toMatchSnapshot()
})
