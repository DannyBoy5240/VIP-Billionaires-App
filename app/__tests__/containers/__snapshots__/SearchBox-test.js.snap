// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#858585",
      "borderRadius": 14,
      "borderStyle": "solid",
      "borderWidth": 0.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "fontSize": 17,
          "marginHorizontal": 8,
          "marginVertical": 4,
          "paddingHorizontal": 10,
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#858585",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": undefined,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï”…
    </Text>
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      blurOnSubmit={true}
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="#C4C4C4"
      returnKeyType="search"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "height": 40,
            "marginLeft": 8,
            "paddingBottom": 0,
            "paddingTop": 0,
          },
          Object {
            "color": "#FFFFFF",
          },
        ]
      }
      theme="dark"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
